evil_hangman.py

```diff
@@ -0,0 +1,157 @@
+    #   at the beginning. As letters are matched, word is re-built
+    dictoptions = {}
+    # This for loop goes over all words in the current list and
+    # classifies them according to the match pattern
+    for word in currentlist:
+        pattern = list(wip)
+        # look_for_match will return a list with the indexes in which
+        # letter is found in the word
+        indexes = look_for_match(letter, word.upper())
+        pattern = update_pattern_with_letter_match(letter, indexes, pattern)
+        # convert the pattern stored on a list into a string to use as
+        # key to the dictionary listoptions
+        pattern_string = ''.join(pattern)
+        if pattern_string in dictoptions:
+                dictoptions[pattern_string].append(word)
+        else:
+            dictoptions[pattern_string] = [word]
+    # In the unlikely scenario that there are no more new options
+    # to create a new list of words, this function returns the
+    # current word_in_process and the current list
+    # But if there are more options to create new word lists, it returns
+    # the list with the most words and the corresponding pattern to that list
+    if len(dictoptions) == 0:
+        return(currentlist, thelist)
+    else:
+        return max_list(dictoptions)
+
+
+if __name__ == '__main__':
+    print_greeting()
+    print()
+    play = True
+    # Reads dictionary of words
+    file_location = open("/usr/share/dict/words")
+    file_content = file_location.read()
+    words = file_content.split()
+    file_location.close()
+    # Checks to see what the longest word is so that a random length
+    # can be selected without selecting a number larger than the
+    # largest word
+    max_word_length = 1
+    for word in words:
+        if len(word) > max_word_length:
+            max_word_length = len(word)
+    # Game Begins
+    while play is True:
+        random_length = randint(1, max_word_length)
+    # Initializes global variables
+        tries = 0
+        success = False
+        characters_read = []
+        word_in_process = ['-']*random_length
+    # Generates the initial word list made out of all words of the
+    # word length generated
+        word_list = [word for word in words if len(word) == random_length]
+    # Tells the player how long the word is
+        print("Your word has {} letters".format(random_length))
+    # Game will ask for a letter giving the player 8 tries
+        while success is not True and tries < 8:
+            letter = get_letter(characters_read)
+            evil_next_step = evil_new_list(letter, word_list, word_in_process)
+            word_in_process = list(evil_next_step[0])
+            word_list = list(evil_next_step[1])
+            print_word_status(word_in_process, tries)
+            if ''.join(word_in_process).count('-') == 0:
+                print("wow! you won!!")
+                success = True
+            tries += 1
```

Comment:
The number of tries should not increment if the player successfully guessed a character, same as in normal play.